lib/commands/project.ts

```diff
@@ -88,38 +206,90 @@ export class CreateNativeCommand extends ProjectCommandBase {
 }
 $injector.registerCommand("create|native", CreateNativeCommand);
 
-export class InitHybridCommand extends ProjectCommandBase {
+export class InitCommand extends InitProjectCommandBase {
+	constructor($project: Project.IProject,
+		$errors: IErrors,
+		$fs: IFileSystem,
+		$logger: ILogger) {
+		super($project, $errors, $fs, $logger);
+	}
+
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			if(this.isCordovaProject().wait()) {
+				this.$logger.info("Trying to create hybrid project.");
+				this.createProjectFileFromExistingProject(projectTypes.Cordova).wait();
+			} else if(this.isNativeProject().wait()) {
+				this.$logger.info("Trying to create native project.");
+				this.createProjectFileFromExistingProject(projectTypes.NativeScript).wait();
+			} else if(this.isWebsiteProject().wait()) {
+				this.$logger.info("Trying to create mobile website project.");
+				this.createProjectFileFromExistingProject(projectTypes.MobileWebsite).wait();
+			} else {
+				this.$errors.fail("Cannot determine project type. Please specify if the project is hybrid, native or mobile website.");
+			}
+		}).future<void>()();
+	}
+}
+$injector.registerCommand("init|*unknown", InitCommand);
+
+export class InitHybridCommand extends InitProjectCommandBase {
 	constructor($project: Project.IProject,
-		$errors: IErrors) {
-		super($project, $errors);
+		$errors: IErrors,
+		$fs: IFileSystem,
+		$logger: ILogger) {
+		super($project, $errors, $fs, $logger);
 	}
 
 	public execute(args: string[]): IFuture<void> {
-		return this.createProjectFileFromExistingProject(projectTypes.Cordova);
+		return (() => {
+			if(this.isCordovaProject().wait()) {
+				this.createProjectFileFromExistingProject(projectTypes.Cordova).wait();
+			} else {
+				this.$errors.fail({ formatStr: "The specified folder is not a valid hybrid project. You can check if it contains tns_modules dir or app/bootstrap.js (both are used for NativeScript projects).", suppressCommandHelp: true });
+			}
+		}).future<void>()();
 	}
 }
 $injector.registerCommand("init|hybrid", InitHybridCommand);
 
-export class InitNativeCommand extends ProjectCommandBase {
+export class InitNativeCommand extends InitProjectCommandBase {
 	constructor($project: Project.IProject,
-		$errors: IErrors) {
-		super($project, $errors);
+		$errors: IErrors,
+		$fs: IFileSystem,
+		$logger: ILogger) {
+		super($project, $errors, $fs, $logger);
 	}
 
 	public execute(args: string[]): IFuture<void> {
-		return this.createProjectFileFromExistingProject(projectTypes.NativeScript);
+		return (() => {
+			if(this.isNativeProject().wait()) {
+				this.createProjectFileFromExistingProject(projectTypes.NativeScript).wait();
+			} else {
+				this.$errors.fail({ formatStr: "The specified folder is not a valid NativeScript project. You can check if it contains any cordova.*.js files (used in hybrid projects) or index.html file (used in hybrid projects and mobile websites).", suppressCommandHelp: true });
```

Comment:
I'll remove this check and the error message, if the user specifies project type (native, hybrid, website) we will assume he knows what to do, so no validation here.